### ISSUE TITLE
Fails when source code contains two type names that differ only in case

### ISSUE THREAD

**Issue by svver** [dotnet/docfx#219]
We have a FooClass and a fooClass in the same namespace 

```
namespace FooLib
{
    /// <summary>
    /// Foo class in Pascal-style
    /// </summary>
    public class FooClass {}

    /// <summary>
    /// Foo class in camel case
    /// </summary>
    public class fooClass {}
}
```

docfx 1.5.2 build step fails with an error: Unable to find file with uid "FooLib.FooClass" referenced by TOC file "C:\docFx\test\docfx_project\api\toc.yml"

docfx 1.6.1 build step fails with an error: Error transforming model "obj\rawmodel\api\FooLib.raw.json" generated from "C:\docFx\test\docfx_project\api\FooLib.yml" using "ManagedReference.html.primary.tmpl.js", and a warning: Warning: [Build Document](file:///C:/docFx/test/docfx_project/api/toc.yml)Unable to find file with uid "FooLib.FooClass" referenced by TOC file "C:\docFx\test\docfx_project\api\toc.yml"

Seems that the problem is that docfx uses a type full name as a .yml file name while file names a case insensitive. 

I understand that this kind of class naming violates even the common sense, but as it is valid in C# it should be valid in docfx.

**Comment by gboulard** [comment]
interesting in that bugfix !
i got 2 classes that are auto-generated with case-sensitive only differences.
will it be fixed in 2.19 ?

**Comment by vicancy** [comment]
https://github.com/dotnet/docfx/pull/1777 will be fixed in 2.20 @gboulard
